src/main/java/pl/patrycja/ox/gameflow/Mode.java

```diff
@@ -1,41 +1,54 @@
 package pl.patrycja.ox.gameflow;
 
 import pl.patrycja.ox.GameSettings;
+import pl.patrycja.ox.Sign;
 import pl.patrycja.ox.ui.InputChecker;
 import pl.patrycja.ox.ui.UI;
 import pl.patrycja.ox.winnerchecker.Spectator;
-import pl.patrycja.ox.winnerchecker.SpectatorsRoom;
+import pl.patrycja.ox.winnerchecker.Spectators;
 
 import java.util.List;
 
 abstract class Mode {
 
     protected GameSettings gameSettings;
     protected UI ui;
+    private List<Spectator> spectatorsList;
 
     public Mode(UI ui) {
         this.ui = ui;
     }
 
     void play(List<Player> players) {
-        gameSettings.setPlayer();
-        PlayerChanger playerChanger = new PlayerChanger(players, gameSettings);
-
-        for (int i = 0; i < gameSettings.getNumberOfMatches(); i++) {
-            List<Spectator> spectators = SpectatorsRoom.addSpectators(gameSettings);
-            //TODO: changes argument gameSetting to boardSize
-            Match.init(gameSettings, spectators)
-                    .addController(playerChanger)
-                    .start();
+        Player initialPlayer = askWhichPlayerStarts(players);
+
+        for (int i = 1; i <= gameSettings.getNumberOfMatches(); i++) {
+            Match.init(gameSettings.getBoardSize(), ui, spectatorsList)
+                    .addPlayers(players, initialPlayer)
+                    .start(i);
         }
     }
 
+    void createSpectators() {
+        Spectators spectators = new Spectators(gameSettings, ui);
+        spectatorsList = spectators.create();
+    }
+
     abstract List<Player> createPlayers();
 
     abstract void settings(String[] inputArrayParameters);
 
-    protected void checkIfCorrectInputData(String[] inputArrayParameters) {
+    void checkIfCorrectInputData(String[] inputArrayParameters) {
+        InputChecker inputChecker = new InputChecker(ui);
+        inputChecker.checkIfInputParametersAreInteger(inputArrayParameters);
+        inputChecker.checkIfUnbrokenLineIsMoreThanBoardSize(inputArrayParameters);
+    }
+
+    Player askWhichPlayerStarts(List<Player> players) {
         InputChecker inputChecker = new InputChecker(ui);
-        inputChecker.checkIfInputParametersAreValid(inputArrayParameters);
+        ui.display("Which player should start: O or X ? ");
+        Sign sign = inputChecker.checkSignValidate(ui.read());
+        //noinspection OptionalGetWithoutIsPresent player always be there
```

Comment:
Similar here, if there will be more signs this will not be true, can be skipped for now. Discussed offline.